src/drivers/imu/mpu9250/mag.cpp

```diff
@@ -586,40 +607,54 @@ MPU9250_mag::ak8963_setup(void)
 
 		retries--;
 		PX4_ERR("AK8963: bad id %d retries %d", id, retries);
-		_parent->modify_reg(MPUREG_USER_CTRL, 0, BIT_I2C_MST_RST);
-		up_udelay(100);
+		_parent->modify_reg(AK_MPU_OR_ICM(MPUREG_USER_CTRL, ICMREG_20948_USER_CTRL), 0, BIT_I2C_MST_RST);
+		usleep(100);
```

Comment:
usleep doesn't work in interrupt context - in which context is this running?

Reply:
This should not  be called in interrupt context - it's run on driver start and in the event of a reset during ioctl or in an error event during polling. I'll check if I can take some of those out however.

Reply:
I would change that back to up_udelay() here.